From 06e32182a639fbc912b4b1f70a6e2386fd23d6b1 Mon Sep 17 00:00:00 2001
From: "David R. Bild" <david.bild@xaptum.com>
Date: Sat, 23 Jun 2018 02:00:12 +0000
Subject: [PATCH 2/2] mmc/sdhci-of-at91: mark HS200 mode as broken

The tuning procedure required by the eMMC HS200 mode occasionally
fails on the XAP-RC-001 router cards. So mark that mode as broken, so
that DDR52, which doesn't require tuning, is used instead.
---
 drivers/mmc/host/sdhci-of-at91.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/mmc/host/sdhci-of-at91.c b/drivers/mmc/host/sdhci-of-at91.c
index 4e47ed6bc716..737214f133ee 100644
--- a/drivers/mmc/host/sdhci-of-at91.c
+++ b/drivers/mmc/host/sdhci-of-at91.c
@@ -364,6 +364,14 @@ static int sdhci_at91_probe(struct platform_device *pdev)
 	pm_runtime_set_autosuspend_delay(&pdev->dev, 50);
 	pm_runtime_use_autosuspend(&pdev->dev);
 
+	/*
+	 * The tuning procedure required by the eMMC HS200 mode
+	 * occasionally fails on the XAP-RC-001 router cards. So mark
+	 * that mode as broken, so that DDR52, which doesn't require
+	 * tuning, is used instead.
+	 */
+	host->quirks2 |= SDHCI_QUIRK2_BROKEN_HS200;
+
 	ret = sdhci_add_host(host);
 	if (ret)
 		goto pm_runtime_disable;
-- 
2.11.0

